sdk/eventgrid/azure-messaging-eventgrid/src/main/java/com/azure/messaging/eventgrid/EventGridSasGenerator.java

```diff
@@ -20,32 +18,54 @@
 import java.util.Base64;
 
 /**
- * A way to use a generated shared access signature as a credential to publish events to a topic through a client.
+ * This is a utility to create a SAS, or Shared Access Signature token, from the endpoint and key of an
+ * Event Grid Topic or Domain.
+ *
+ * If you would like to grant somebody to publish events to your Event Grid Topic/Domain resource for a limited time,
+ * you could use this utility class to create a SAS token and share the token to them. They can then use
+ * {@link EventGridPublisherClientBuilder#credential(AzureSasCredential)} to to create a client to publish events.
+ *
  */
-public final class EventGridSasCredential {
-
-    private String sas;
+public final class EventGridSasGenerator {
+    private EventGridSasGenerator() {
+        // Hide the constructor
+    }
+    private static final ClientLogger logger = new ClientLogger(EventGridPublisherClient.class);
 
-    private static final ClientLogger logger = new ClientLogger(EventGridSasCredential.class);
+    /**
+     * Generate a shared access signature to provide time-limited authentication for requests to the Event Grid
+     * service with the latest Event Grid service API defined in {@link EventGridServiceVersion#getLatest()}.
+     * @param endpoint the endpoint of the Event Grid topic or domain.
+     * @param expirationTime the time in which the signature should expire, no longer providing authentication.
+     * @param keyCredential  the access key obtained from the Event Grid topic or domain.
+     *
+     * @return the shared access signature string which can be used to construct an instance of
+     * {@link AzureSasCredential}.
+     */
+    public static String generateSas(String endpoint, AzureKeyCredential keyCredential, OffsetDateTime expirationTime) {
+        return generateSas(endpoint, keyCredential, expirationTime, EventGridServiceVersion.getLatest());
+    }
 
     /**
      * Generate a shared access signature to provide time-limited authentication for requests to the Event Grid
      * service.
-     * @param endpoint       the endpoint of the Event Grid topic or domain.
+     * @param endpoint the endpoint of the Event Grid topic or domain.
+     * @param apiVersion the EventGrid service api version defined in {@link EventGridServiceVersion}
```

Comment:
The order of @param should be the same as the order in the method signature.

Reply:
Good catch. Corrected.